core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -51,6 +52,21 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
+  @GET
+  @Path("executors/{executorId}/threads")
+  def threadDump(@PathParam("executorId") executorId: String):
+  Option[Array[ThreadStackTrace]] = withUI { ui =>
+    val safeExecutorId =
+      Option(UIUtils.stripXSS(executorId)).map { executorId =>
+        UIUtils.decodeURLParameter(executorId)
+      }.getOrElse {
+        throw new IllegalArgumentException(s"Missing executorId parameter")
+      }
+    ui.sc.flatMap { sc =>
```

Comment:
Should an error be thrown if this is called in a replayed app?

GONE/410 sounds like the most appropriate.